pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,199 +1,37 @@
-    name = "${finalAttrs.pname}-cache-${finalAttrs.version}";
-  };
-
-  strictDeps = true;
-
-  nativeBuildInputs =
-    [
-      ncurses
-      pandoc
-      pkg-config
-      removeReferencesTo
-      zig_hook
-    ]
-    ++ lib.optionals (appRuntime == "gtk") [
-      glib # Required for `glib-compile-schemas`
-      wrapGAppsHook4
-    ];
-
-  buildInputs =
-    [
-      glslang
-      oniguruma
-    ]
-    ++ lib.optional (appRuntime == "gtk" && withAdwaita) libadwaita
-    ++ lib.optional (appRuntime == "gtk") libX11
-    ++ lib.optional (renderer == "opengl") libGL
-    ++ lib.optionals (fontBackend == "fontconfig_freetype") [
-      bzip2
-      fontconfig
-      freetype
-      harfbuzz
-    ];
-
-  zigBuildFlags =
-    [
-      "--system"
-      "${finalAttrs.deps}"
-      "-Dversion-string=${finalAttrs.version}"
-
-      "-Dapp-runtime=${appRuntime}"
-      "-Dfont-backend=${fontBackend}"
-      "-Dgtk-adwaita=${lib.boolToString withAdwaita}"
-      "-Drenderer=${renderer}"
-    ]
-    ++ lib.mapAttrsToList (name: package: "-fsys=${name} --search-prefix ${lib.getLib package}") {
-      inherit glslang;
-    };
-
-  zigCheckFlags = finalAttrs.zigBuildFlags;
-
-  doCheck = true;
-
-  /**
-    Ghostty really likes all of it's resources to be in the same directory, so link them back after we split them
-
-    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/os/resourcesdir.zig#L11-L52
-    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L745-L750
-    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L818-L834
-
-    terminfo and shell integration should also be installable on remote machines
-
-    ```nix
-    { pkgs, ... }: {
-      environment.systemPackages = [ pkgs.ghostty.terminfo ];
-
-      programs.bash = {
-        interactiveShellInit = ''
-          if [[ "$TERM" == "xterm-ghostty" ]]; then
-            builtin source ${pkgs.ghostty.shell_integration}/bash/ghostty.bash
-          fi
-        '';
-      };
-    }
-    ```
-  */
-  postFixup = ''
-    ln -s $man/share/man $out/share/man
-
-    moveToOutput share/terminfo $terminfo
-    ln -s $terminfo/share/terminfo $out/share/terminfo
-
-    mv $out/share/ghostty/shell-integration $shell_integration
-    ln -s $shell_integration $out/share/ghostty/shell-integration
-
-    mv $out/share/vim/vimfiles $vim
-    rmdir $out/share/vim
-    ln -s $vim $out/share/vim-plugins
-
-
-    remove-references-to -t ${finalAttrs.deps} $out/bin/.ghostty-wrapped
-  '';
-
-  nativeInstallCheckInputs = [
-    versionCheckHook
-  ];
-
-  doInstallCheck = true;
-
-  versionCheckProgramArg = [ "--version" ];
-
-  passthru = {
-    tests = lib.optionalAttrs stdenv.hostPlatform.isLinux {
-      inherit (nixosTests) allTerminfo;
-      nixos = nixosTests.terminal-emulators.ghostty;
+if settings == { } then
+  ghostty-unwrapped
+else
```

Comment:
I am not sure we want this shortcut here. It could affect bugs which are only caused when using the wrapper.